torch/onnx/symbolic_opset14.py

```diff
@@ -117,3 +119,174 @@ def quantized_hardswish(g: jit_utils.GraphContext, x, op_scale, op_zero_point):
+
+    # Swap the last two axes of key
+    key_shape = g.op("Shape", key)
+    end_idx = g.op(
+        "Constant", value_t=torch.tensor([_constants.INT64_MAX], dtype=torch.int64)
+    )
+    last_idx = g.op("Constant", value_t=torch.tensor([-1], dtype=torch.int64))
+    second_last_idx = g.op("Constant", value_t=torch.tensor([-2], dtype=torch.int64))
+    key_last_dim = g.op("Slice", key_shape, last_idx, end_idx)
+    key_second_last_dim = g.op("Slice", key_shape, second_last_idx, last_idx)
+    key_first_dims = g.op(
+        "Slice",
+        key_shape,
+        g.op(
+            "Constant", value_t=torch.tensor([_constants.INT64_MIN], dtype=torch.int64)
+        ),
+        second_last_idx,
+    )
+    # Contract the dimensions that are not the last two so we can transpose
+    # with a static permutation.
+    key_squeezed_shape = g.op(
+        "Concat",
+        g.op("Constant", value_t=torch.tensor([-1], dtype=torch.int64)),
+        key_second_last_dim,
+        key_last_dim,
+        axis_i=0,
+    )
+    key_squeezed = g.op("Reshape", key, key_squeezed_shape)
+    key_squeezed_transposed = g.op("Transpose", key_squeezed, perm_i=[0, 2, 1])
+    key_transposed_shape = g.op(
+        "Concat", key_first_dims, key_last_dim, key_second_last_dim, axis_i=0
+    )
+    key_transposed = g.op("Reshape", key_squeezed_transposed, key_transposed_shape)
+
+    # https://github.com/pytorch/pytorch/blob/12da0c70378b5be9135c6fda62a9863bce4a4818/aten/src/ATen/native/transformers/attention.cpp#L653
+    # Scale q, k before matmul for stability see https://tinyurl.com/sudb9s96 for math
+    query_scaled = g.op("Mul", query, g.op("Sqrt", scale))
+    key_transposed_scaled = g.op("Mul", key_transposed, g.op("Sqrt", scale))
+    mul_qk = g.op("MatMul", query_scaled, key_transposed_scaled)
+
+    if symbolic_helper._is_none(attn_mask):
+        mul_qk_add = mul_qk
+    elif (
+        _type_utils.JitScalarType.from_value(attn_mask)
+        == _type_utils.JitScalarType.BOOL
+    ):
+        # Turn the Boolean mask to float: attn_mask.masked_fill(not attn_mask, -float('inf'))
+        const_zero = g.op("Constant", value_t=torch.tensor([0.0]))
+        const_neg_inf = g.op("Constant", value_t=torch.tensor([-float("inf")]))
+        attn_mask = g.op("Where", attn_mask, const_zero, const_neg_inf)
+        mul_qk_add = g.op("Add", mul_qk, attn_mask)
+    elif (
+        _type_utils.JitScalarType.from_value(attn_mask)
+        == _type_utils.JitScalarType.FLOAT
+    ):
+        mul_qk_add = g.op("Add", mul_qk, attn_mask)
+    else:
+        raise ValueError(
+            "Unsupported type for attn_mask: {}".format(
+                _type_utils.JitScalarType.from_value(attn_mask)
+            )
+        )
+
+    attn_weight = g.op("Softmax", mul_qk_add, axis_i=-1)
+    attn_weight, _ = g.op(
+        "Dropout",
+        attn_weight,
+        g.op("Constant", value_t=torch.tensor(dropout_p, dtype=torch.float)),
+        outputs=2,
+    )
```

Comment:
```suggestion
    attn_weight = g.op(
        "Dropout",
        attn_weight,
        g.op("Constant", value_t=torch.tensor(dropout_p, dtype=torch.float)),
    )
```

Dropout can have one or two outputs. We set two outputs in the onnxscript version because it was a limitation in onnxscript eager mode

Reply:
We can also condition on dropout_p != 0 and conditionally add the dropout node here

Reply:
Done